src/test/java/com/google/cloud/spanner/pgadapter/python/django/DjangoTestSetup.java

```diff
@@ -16,15 +16,139 @@
 
+import com.google.cloud.spanner.MockSpannerServiceImpl.StatementResult;
+import com.google.cloud.spanner.Statement;
+import com.google.protobuf.ListValue;
+import com.google.protobuf.Value;
+import com.google.spanner.v1.ResultSet;
+import com.google.spanner.v1.ResultSetMetadata;
+import com.google.spanner.v1.StructType;
+import com.google.spanner.v1.StructType.Field;
+import com.google.spanner.v1.Type;
+import com.google.spanner.v1.TypeCode;
 import java.io.File;
 import java.io.IOException;
 import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.List;
 import java.util.Scanner;
+import org.junit.BeforeClass;
 
 public class DjangoTestSetup extends DjangoMockServerTest {
 
+  public static ResultSet createResultSet(List<String> rows, int numCols) {
+    ResultSet.Builder resultSetBuilder = ResultSet.newBuilder();
+
+    StructType.Builder builder = StructType.newBuilder();
+
+    for (int i = 0; i < 2 * numCols; i += 2) {
+      String colType = rows.get(i);
+      if (colType.equals("int")) {
+        builder.addFields(
+            Field.newBuilder()
+                .setType(Type.newBuilder().setCode(TypeCode.INT64).build())
+                .setName(rows.get(i + 1))
+                .build());
+
+      } else if (colType.equals("string")) {
+        builder.addFields(
+            Field.newBuilder()
+                .setType(Type.newBuilder().setCode(TypeCode.STRING).build())
+                .setName(rows.get(i + 1))
+                .build());
+      } else if (colType.equals("array")) {
+        builder.addFields(
+            Field.newBuilder()
+                .setType(
+                    Type.newBuilder()
+                        .setCode(TypeCode.ARRAY)
+                        .setArrayElementType(Type.newBuilder().setCode(TypeCode.STRING))
+                        .build())
+                .setName(rows.get(i + 1))
+                .build());
+      }
+    }
+    resultSetBuilder.setMetadata(
+        ResultSetMetadata.newBuilder().setRowType(builder.build()).build());
+    for (int i = 2 * numCols; i < rows.size(); i += numCols) {
+      ListValue.Builder rowBuilder = ListValue.newBuilder();
+      for (int j = i; j < i + numCols; ++j) {
+        if (rows.get(j).charAt(0) == '[') {
+          String[] array = rows.get(j).substring(1, rows.get(j).length() - 1).split(",");
+
+          ListValue.Builder listValueBuilder = ListValue.newBuilder();
+          for (String arrElement : array) {
+            listValueBuilder.addValues(Value.newBuilder().setStringValue(arrElement).build());
+          }
+          Value value = Value.newBuilder().setListValue(listValueBuilder.build()).build();
+          rowBuilder.addValues(value);
+        } else rowBuilder.addValues(Value.newBuilder().setStringValue(rows.get(j)).build());
+      }
+      resultSetBuilder.addRows(rowBuilder.build());
+    }
+    return resultSetBuilder.build();
+  }
+
+  @BeforeClass
+  public static void mockUnnecessarySql() {
```

Comment:
nit: Could we rename this to something like `mockDriverInternalSql`? These are actual queries that Django will execute, and without the additional common table expressions for `pg_namespace` and `pg_type`, Django would not work.

Reply:
Done. And since these tests were passing on actual database also, it looks like PG Adapter is handling these statements well enough as of now.